core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/udf/pythonV2/source/RedditSearchSourceOpDesc.java

```diff
@@ -0,0 +1,161 @@
+                "        \n" +
+                "        if len(self.client_secret) == 0:\n" +
+                "            raise ValueError(\"Client Secret cannot be None.\")\n" +
+                "        \n" +
+                "        if len(self.query) == 0:\n" +
+                "            raise ValueError(\"Query cannot be None.\")\n" +
+                "        \n" +
+                "        if self.limit <= 0 or self.limit > 1000:\n" +
+                "            raise ValueError(\"Limit should be larger than 0 and no more than 1000.\")\n" +
+                "        if self.sorting == \"none\":\n" +
+                "            submissions = redditInstance.subreddit(\"all\").search(query=self.query, limit=self.limit)\n" +
+                "        else:\n" +
+                "            submissions = redditInstance.subreddit(\"all\").search(query=self.query, limit=self.limit, sort=self.sorting)\n" +
+                "        for submission in submissions:\n" +
+                "            author = submission.author\n" +
+                "            subreddit = str(submission.subreddit.display_name)\n" +
+                "            edited = None\n" +
+                "            if type(submission.edited) != type(True):\n" +
+                "                edited = datetime.fromtimestamp(submission.edited)\n" +
+                "            tuple_submission = Tuple({\n" +
+                "                \"id\": submission.id,\n" +
+                "                \"name\": submission.name,\n" +
+                "                \"title\": submission.title,\n" +
+                "                \"created_utc\": datetime.fromtimestamp(submission.created_utc),\n" +
+                "                \"edited\": edited,\n" +
+                "                \"is_self\": submission.is_self,\n" +
+                "                \"selftext\": submission.selftext,\n" +
+                "                \"over_18\": submission.over_18,\n" +
+                "                \"is_original_content\": submission.is_original_content,\n" +
+                "                \"locked\": submission.locked,\n" +
+                "                \"score\": submission.score,\n" +
+                "                \"upvote_ratio\": submission.upvote_ratio,\n" +
+                "                \"num_comments\": submission.num_comments,\n" +
+                "                \"permalink\": submission.permalink,\n" +
+                "                \"url\": submission.url,\n" +
+                "                \"author_name\": author.name,\n" +
+                "                \"subreddit\": subreddit\n" +
+                "            })\n" +
+                "            yield tuple_submission";
+        String client_id_real = this.client_id.replace("\n", "").trim();
+        String client_secret_real = this.client_secret.replace("\n", "").trim();
+        String query_real = this.query.replace("\n", "").trim();
```

Comment:
please use camel case format for variable names, this is java code.

Reply:
solved